new chapter with trivy scanning (#290)
acend/container-basics-training: content/en/docs/10/02/_index.md

> Info: You could also predownload this database and point to it during the runtime, if you have this need.

```bash
trivy image quay.io/acend/example-web-python:latest
```

You may have now some outout like this:

```
2022-05-11T11:47:17.787Z        INFO    Need to update DB
2022-05-11T11:47:17.787Z        INFO    DB Repository: ghcr.io/aquasecurity/trivy-db
2022-05-11T11:47:17.787Z        INFO    Downloading DB...
31.76 MiB / 31.76 MiB [------------------------------------------------------------------------------------------------------] 100.00% 10.19 MiB p/s 3.3s
2022-05-11T11:47:26.485Z        INFO    Detected OS: alpine
2022-05-11T11:47:26.485Z        INFO    Detecting Alpine vulnerabilities...
2022-05-11T11:47:26.487Z        INFO    Number of language-specific files: 1
2022-05-11T11:47:26.487Z        INFO    Detecting python-pkg vulnerabilities...

quay.io/acend/example-web-python:latest (alpine 3.15.4)
=======================================================
Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)

Python (python-pkg)
===================
Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)
```

> No issues over here (as this text was written). So let's check another image:

Reviewer: ```suggestion
No issues were detected in this image (at time of writing). So let's check another image:
```

```bash
trivy image debian:stable-slim
```

As you can see, the output is now much larger than before. You could use Trivy in pipelines where it can break the build if the image has serious severities like HIGH or CRITICAL. For more information check the `trivy --help` page.

> ## Scan Filesystem for Vulnerabilities, Secrets and Misconfigurations

Reviewer: ```suggestion
## Scan filesystem for vulnerabilities, secrets and misconfigurations
```

> You are also able to scan projects locally on disc for several things. This scan check your code for best practises and gives you hints about configurations issues / mistakes which can help you to improve your codebase.

Reviewer: ```suggestion
You can also scan projects locally on disc for several things. This scan checks your code for best practices and gives you hints about configurations issues and mistakes which can help you to improve your codebase.
```

```bash
git clone https://github.com/acend/awesome-apps.git
trivy fs --security-checks vuln,secret,config awesome-apps/
```

```
go/Dockerfile (dockerfile)
==========================
Tests: 23 (SUCCESSES: 23, FAILURES: 0, EXCEPTIONS: 0)
Failures: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)

python/Dockerfile (dockerfile)
==============================
Tests: 23 (SUCCESSES: 23, FAILURES: 0, EXCEPTIONS: 0)
Failures: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)
```

> For more informations checkout the [Trivy Github page](https://github.com/aquasecurity/trivy).

Reviewer: ```suggestion
For more information, check out the [Trivy Github page](https://github.com/aquasecurity/trivy).
```
